rfd/0109-opsgenie-plugin.md

```diff
@@ -0,0 +1,130 @@
+# Opsgenie Integration RFD
+## Required approvers
+
+Engineering: @r0mant, @marcoandredinis, @hugoShaka
+Product: @klizhentas, @xinding33
+Security: @reedloden, @jentfoo
+
+## What
+
+This RFD proposes a plugin that allows Teleport to integrate with Opsgenie, allowing access requests to show up as alerts in Opsgenie. This plugin will differ from existing plugins by being part of the Teleport binary directly.
+
+## Scope
+
+### Phase 1
+
+The plugin will be run as a standalone application similar to the existing Pagerduty plugin differing only in that it will be part of the Teleport binary.
+
+### Phase 2
+
+The plugin will follow the 'hosted' operations model and will at that point be possible to run as part of the Auth server directly or as a standalone application.
+
+## Success criteria
+Users are able to configure teleport to automatically create alerts in Opsgenie from access requests.
+Alerts were chosen over incidents in Opsgenie as incidents are intended to be used for high priority alerts that indicate a service interruption.
+Users are also able to configure auto approval flows to be met under certain conditions. E.g when a requester is on-call.
+
+## Configuration UX
+
+The plugin will be configured in Teleport's config yaml file. The required fields will be added to a 'plugins' section containing the required information to interact with both Teleport access and the Opsgenie API.
+
+```
+plugins:
```

Comment:
Other plugins currently support more fields
- log level/severity
- log output location
- Auth Server
- Identity or ClientKey/Cert and RootCAs

Should we add them within the `opsgenie`, `plugins` or a top level `teleport` key? 

Given that Identity is already at `opsgenie`, I think we should add the ClientKey/Cert and RootCAs to it as well.
However, it could make sense to add them as some kind of generic plugins configuration (we can always do that later on :+1: )

Reply:
> Was planning for the logging config to be shared with the rest of the teleport config

Something like this?
```yaml
teleport:
  log:
     level: INFO
     output: stderr
plugins:
   opsgenie:
      addr: ...

Reply:
That was what i was thinking.